test/cmd/newapp.sh

```diff
@@ -45,10 +45,21 @@ os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'mysql-persi
 os::cmd::expect_failure_and_text 'oc new-app mysql-persisten mysql' 'mysql as a local directory'
 os::cmd::expect_failure_and_text 'oc new-app --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git' 'as a Git repository URL:  '
 os::cmd::expect_failure_and_text 'oc new-app https://www.google.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-app https://githb.com/openshift/nodejs-e' 'as a Git repository URL:  '
+os::cmd::expect_failure_and_text 'oc new-app https://examplegit.com/openshift/nodejs-e' 'as a Git repository URL:  '
 os::cmd::expect_failure_and_text 'oc new-build --strategy=docker https://192.30.253.113/openshift/ruby-hello-world.git' 'as a Git repository URL:  '
 os::cmd::expect_failure_and_text 'oc new-build https://www.google.com/openshift/nodejs-e' 'as a Git repository URL:  '
-os::cmd::expect_failure_and_text 'oc new-build https://githb.com/openshift/nodejs-e' 'as a Git repository URL:  '
+os::cmd::expect_failure_and_text 'oc new-build https://examplegit.com/openshift/nodejs-e' 'as a Git repository URL:  '
+
+# setting source secret via the --source-secret flag
+os::cmd::expect_success_and_text 'oc new-app https://github.com/openshift/cakephp-ex --source-secret=mysecret -o yaml' 'name: mysecret'
+os::cmd::expect_success_and_text 'oc new-build https://github.com/openshift/cakephp-ex --source-secret=mynewsecret -o yaml' 'name: mynewsecret'
+os::cmd::expect_success_and_text 'oc new-app examples/quickstarts/cakephp-mysql.json --source-secret=mysecret -o yaml' 'name: mysecret'
+os::cmd::expect_success 'oc create -f examples/quickstarts/cakephp-mysql.json'
+os::cmd::expect_success 'oc get template cakephp-mysql-example'
+os::cmd::expect_success_and_text 'oc new-app cakephp-mysql-example --source-secret=mysecret -o yaml' 'name: mysecret'
+os::cmd::expect_success 'oc delete all --selector="label=cakephp-mysql-example"'
+os::cmd::expect_success 'oc delete template cakephp-mysql-example'
```

Comment:
what are you deleting here?  you ran new-app with -o yaml so you never created anything.

Reply:
that said, it would be good to have at least one test that *does* create the objects (and then clean them up), so you can confirm you're not producing invalid objects.